pages/index.js

```diff
@@ -55,14 +65,18 @@ export default function HomePage() {
         <h2>Anstehende Aufgaben:</h2>
         {submittedData.length > 0 && (
           <div className="submitted-data">
-            {submittedData.map((data, index) => (
-              <p key={index}>
-                Was ist zu tun: {data.option} <br />
-                Bis wann erledigt: {data.date}
-              </p>
+            {submittedData.map((data) => (
+              <div key={data.id} className="task-item">
+                <p>
+                  Was ist zu tun: {data.option} <br />
+                  Bis wann erledigt: {data.date}
+                </p>
+                <button onClick={() => handleDelete(data.id)}>Delete</button>
```

Comment:
Each button should have the correct type set. As by default a button is a submit button and you should set for every button the correct type. Here it would be `type="button"`

Reply:
You are right, I added the correct type.